.gitlab/ci/bucket-upload.yml

```diff
@@ -0,0 +1,147 @@
+    else
+      # Modifying ssh config file
+      echo "Host 10.0.*
+        StrictHostKeyChecking no
+        LogLevel ERROR
+        ProxyJump content-build@content-build-lb.demisto.works  # disable-secrets-detection
+       Host content-build-lb.demisto.works
+        Port 43567
+        UserKnownHostsFile /dev/null
+        StrictHostKeyChecking no
+        LogLevel ERROR" >> ~/.ssh/config
+    fi
+  - echo "Open SSH Tunnel"
+  - Tests/scripts/open_ssh_tunnel.sh
+
+.check_user_permissions_to_upload_packs: &check_user_permissions_to_upload_packs
+  - echo "=====Check User Permissions to Upload Packs=====" # if bucket upload and uploading to marketplace-dist
+  - |
+    if [[ -n "${BUCKET_UPLOAD}" || -n "${FORCE_BUCKET_UPLOAD}" ]] && [[ "$GCS_MARKET_BUCKET" == "marketplace-dist" ]]; then
+      CONTENT_LEADERS=$(curl -sS "https://api.github.com/orgs/demisto/teams/content-leaders/members" -H "Authorization: token ${GITHUB_TOKEN}")
+      echo "recieved content leaders"
+      LEADER_NAMES=$(echo $CONTENT_LEADERS | jq -r ".[].login")
+      LEADER_NAMES=$(echo "${LEADER_NAMES}" "content-bot" )
+      if [[ -z "$GITLAB_USER_NAME" ]] || [[ -z "`echo $LEADER_NAMES | grep -w "$GITLAB_USER_NAME"`" ]]; then
+        echo -e "User '$GITLAB_USER_NAME' is not allowed to trigger this build, only one of:\n${LEADER_NAMES}"
+        exit 1
+      else
+        echo "User '${GITLAB_USER_NAME}' is allowed to upload packs / force upload packs."
+      fi
+    fi
+
+.install_packs_in_server:
+  needs: ["create-instances"]
+  stage: run-instances
+  artifacts:
+    expire_in: 48 hrs
+    paths:
+      - /builds/xsoar/content/artifacts/*
+  variables:
+    INSTANCE_CREATED: "true"
+    SSH_TUNNEL_TIMEOUT: 10
+    TIME_TO_LIVE: ""
+  extends:
+    - .bucket-upload-rule
+    - .default-job-settings
+  script:
+    - echo "======= Download configuration ========"
+    - ./Tests/scripts/download_demisto_conf.sh  >> $ARTIFACTS_FOLDER/logs/installations.log
+    - *start_tunnel
+    - echo "=====Get Instance Variables====="
+    - echo INSTANCE_ROLE="$INSTANCE_ROLE"
+    - echo INSTANCE_CREATED="$INSTANCE_CREATED"
+    - echo "=====Wait Until Server Ready====="
+    - python3 ./Tests/scripts/wait_until_server_ready.py "$INSTANCE_ROLE"
+    - echo "=====Install Packs====="
+    - ./Tests/Marketplace/install_packs.sh "$INSTANCE_ROLE"
+    - echo "=====Destroy Instances====="
+    - |
+      if [ -n "${CONTRIB_BRANCH}" ]; then
+        echo "Skipping - not running in contribution instance"
+      else
+        python3 ./Tests/scripts/destroy_instances.py $ARTIFACTS_FOLDER $ENV_RESULTS_PATH "$INSTANCE_ROLE" "$TIME_TO_LIVE"
+        export PSWD=$(jq .serverLogsZipPassword < $(cat secret_conf_path) | cut -d \" -f 2)
+        zip -P $PSWD -j $ARTIFACTS_FOLDER/Logs.zip $ARTIFACTS_FOLDER/server*.log || ((($? > 0)) && echo "Didn’t find any server logs, skipping this stage" && exit 0)
+        rm -f $ARTIFACTS_FOLDER/server*.log
+      fi
```

Comment:
The difference here is `./Tests/Marketplace/install_packs.sh "$INSTANCE_ROLE"` instead of `    - ./Tests/scripts/install_content_and_test_integrations.sh "$INSTANCE_ROLE" || EXIT_CODE=$?
    - ./Tests/scripts/run_tests.sh "$INSTANCE_ROLE"
    - exit $EXIT_CODE`

Reply:
I'll make it more generic though it won't be like you suggested.

Reply:
Honestly the code duplicity is just 
`Tests/scripts/open_ssh_tunnel.sh` and `python3 ./Tests/scripts/wait_until_server_ready.py "$INSTANCE_ROLE"` and `python3 ./Tests/scripts/destroy_instances.py $ARTIFACTS_FOLDER $ARTIFACTS_FOLDER/env_results.json "$INSTANCE_ROLE" "$TIME_TO_LIVE"`

I don't really want to get into changing .test_content_on_server_instances_base just so these three lines won't be duplicated